### PR TITLE
Bug 1802719: oc adm groups sync  IsQueryOutOfBoundsError warning,not error

### PR DESCRIPTION
**Pull request by sallyom**
/cc @soltysh

### CODE REVIEW

**Comment by sallyom** [comment]
spent some time adding a unit and/or extended test for this, will sync w/ @stlaz

File: pkg/helpers/groupsync/groupsyncer.go [hunk 1]
```diff
@@ -85,6 +86,10 @@ func (s *LDAPGroupSyncer) Sync() ([]*userv1.Group, []error) {
 		// update the OpenShift Group corresponding to this record
 		openshiftGroup, err := s.makeOpenShiftGroup(ldapGroupUID, usernames)
 		if err != nil {
+			if ldapquery.IsQueryOutOfBoundsError(err) {
+				klog.Warningf("%s", err.Error())
```
Comment (soltysh) [inline pkg/helpers/groupsync/groupsyncer.go:90]: We rarely use `klog` for these warnings, `fmt.Fprintf(os.Stderr` is the preferred.

**Comment by stlaz** [comment]
/hold
why sync groups that are outside the intended tree?

File: pkg/helpers/groupsync/groupsyncer.go (hunk 1 above)
Comment (sallyom) [reply pkg/helpers/groupsync/groupsyncer.go]: updated

**Comment by stlaz** [comment]
/hold cancel
nvm, you're not actually syncing it, my bad

**Pull request merged**